src/main/java/uk/nhs/ctp/service/ReferralRequestService.java

```diff
@@ -1,58 +1,57 @@
 package uk.nhs.ctp.service;
 
-import ca.uhn.fhir.context.FhirContext;
-import ca.uhn.fhir.parser.IParser;
-import com.google.common.base.Preconditions;
-import java.util.function.Consumer;
-import javax.transaction.Transactional;
+import static java.util.Collections.singletonList;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 import lombok.AllArgsConstructor;
-import org.hl7.fhir.dstu3.model.IdType;
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Bundle.BundleEntryComponent;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.Reference;
 import org.hl7.fhir.dstu3.model.ReferralRequest;
+import org.hl7.fhir.dstu3.model.ResourceType;
 import org.springframework.stereotype.Service;
-import uk.nhs.ctp.entities.ReferralRequestEntity;
-import uk.nhs.ctp.repos.ReferralRequestRepository;
+import uk.nhs.ctp.service.dto.SelectedServiceRequestDTO;
 import uk.nhs.ctp.service.fhir.ReferenceService;
-import uk.nhs.ctp.transform.ReferralRequestEntityTransformer;
-import uk.nhs.ctp.utils.ErrorHandlingUtils;
+import uk.nhs.ctp.service.fhir.StorageService;
+import uk.nhs.ctp.utils.RetryUtils;
 
 @Service
 @AllArgsConstructor
+@Slf4j
 public class ReferralRequestService {
 
-  private ReferralRequestEntityTransformer referralRequestEntityTransformer;
-  private ReferralRequestRepository referralRequestRepository;
-  private FhirContext fhirContext;
+  private AppointmentService appointmentService;
+  private StorageService storageService;
   private ReferenceService referenceService;
 
-  public ReferralRequest makeAbsolute(ReferralRequest referralRequest) {
-    IdType idElement = referralRequest.getIdElement();
-    Preconditions.checkArgument(idElement.isAbsolute(), "Referral request must have absolute ID");
-    String baseUrl = idElement.getBaseUrl();
-
-    ReferralRequest copy = referralRequest.copy();
-    copy.setIdElement(null);
-    referenceService.resolveRelative(baseUrl, copy);
-
-    return copy;
+  public void updateServiceRequested(SelectedServiceRequestDTO requestDTO) {
+    log.info("Setting selected HealthcareService for case " + requestDTO.getCaseId());
+    ReferralRequest referralRequest = getByCaseId(requestDTO.getCaseId()).orElseThrow();
```

Comment:
I feel like updating the service should refer to a specific RR, not just whichever one we find.
The services returned by the DoS will be specific to a particular RR. What if the original RR has been cancelled and a new active RR is present - the selected service would be wrong.

Ideally the SelectedServiceRequestDTO would include the reference of the RR which triggered the search, and the status of that RR checked to make sure it's still valid to select the service at this point.

Reply:
i'll raise a tech debt to fix this